taskvine/src/manager/vine_manager.c

```diff
@@ -343,6 +343,44 @@ static vine_msg_code_t handle_info(struct vine_manager *q, struct vine_worker_in
 	return VINE_MSG_PROCESSED;
 }
 
+/*
+At this point we have received a cache update for a temp file. If q->temp_replica_count > 0 then tell up to that
+many workers to get the file from worker w.
+*/
+static void replicate_temp_file(
+		struct vine_manager *q, struct vine_worker_info *w, const char *cachename, long long size)
+{
+	if (!q->temp_replica_count)
+		return;
+
+	int found = 0;
+	struct vine_worker_info **workers;
+	workers = vine_file_replica_table_find_replication_targets(q, w, cachename, &found);
+
+	debug(D_VINE, "Found %d available workers to replicate %s", found, cachename);
+
+	for (int i = 0; i < found; i++) {
+		struct vine_worker_info *peer = workers[i];
+		char *worker_source = string_format("worker://%s:%d/%s", w->transfer_addr, w->transfer_port, cachename);
+		char *worker_addr = string_format("worker://%s:%d", w->transfer_addr, w->transfer_port);
+
+		char *transfer_id = vine_current_transfers_add(q, peer, worker_source);
```

Comment:
For consistency, break this out into `vine_manager_put_url_now` next to `vine_manager_put_url`.
Also need to url encode the names.